### PR TITLE
Don't store tensors in init args

### PR DESCRIPTION
**Pull request by peastman**
On some computers (but not others) this leads to an error when resuming training, due to tensors stored in `hparams.yaml`.  I'm not sure why it only happens on some computers.

### CODE REVIEW

**Comment by PhilippThoelke** [comment]
Any reason you used the `.numpy()` list comprehension and `float()` over `tensor.tolist()` and `tensor.item()`? The latter are safer when tensors are on different devices.

**Comment by peastman** [comment]
Good idea.  I just changed it.

**Comment by peastman** [comment]
All tests are now passing.

I also fixed a second error that prevented resuming training when using D2.  The constructor received the scale factors from arguments or from the dataset, then converted them to different units.  That meant `get_init_args()` would return the converted values rather than the ones that were actually passed in.  When you tried to resume training, it would therefore receive the wrong values, which lead to an exception.

File: torchmdnet/priors/zbl.py [hunk 1]
```diff
@@ -22,7 +22,7 @@ def __init__(self, cutoff_distance, max_num_neighbors, atomic_number=None, dista
             distance_scale = dataset.distance_scale
         if energy_scale is None:
             energy_scale = dataset.energy_scale
-        atomic_number = torch.as_tensor(atomic_number, dtype=torch.int8)
+        atomic_number = torch.as_tensor(atomic_number, dtype=torch.long)
```
Comment (stefdoerr) [inline torchmdnet/priors/zbl.py:25]: is there a reason we need to store atomic numbers as int64? Seems excessive

**Review by stefdoerr — COMMENTED** [review]
Beyond my comment on the torch.long, it looks good to me

File: torchmdnet/priors/zbl.py (hunk 1 above)
Comment (peastman) [reply torchmdnet/priors/zbl.py:25]: Using int8 led to exceptions on the GPU.  This is just a single array whose length equals the number of atom types (generally less than 100).  The storage is negligible.

**Comment by peastman** [comment]
> Beyond my comment on the torch.long, it looks good to me

Can you submit a review approving it?  I can't merge it until you do.

**Review by stefdoerr — APPROVED** [review]

**Comment by peastman** [comment]
Thanks!

**Pull request merged**